### DISCUSSION TITLE
Add Electron options from the `Window` facade

### DISCUSSION THREAD

**Discussion by lightwalkernet** [NativePHP/laravel#282] [Ideas]
On the window facade it would be helpful to have access to be able to add electron options such as webPreferences as in the example below. This is helpful with opening external websites inside of the app.

`webPreferences: { nodeIntegration: false,
      contextIsolation: false,
      enableRemoteModule: true, }`

Currently I don't see an endpoint in the WindowManger to allow for additional settings and options to be added to the Electron BrowserWindow

**Comment by arnor01** [comment]
I would like to see this as well to be able to load local files directly in an img tag with `file:///`.

**Comment by simonhamp** [comment]
This isn't supported yet, but could be! `webPreferences` are just a JS object. Happy to review a PR that exposes this.

**Comment by simonhamp** [comment]
You can [now](https://github.com/NativePHP/laravel/pull/391) pass `webPreferences` via the `Window::webPreferences()` method.
